### ISSUE TITLE
SourceID doesn't work as SourceTokenOrExistingSourceId did in 19.x

### ISSUE THREAD

**Issue by flyer299** [stripe/stripe-dotnet#1374]
I had code that used to look up a Card ID from a customer's save card list.  Then I would Create a new charge by passing the Card ID into:  `SourceTokenOrExistingSourceId`.

Now `SourceTokenOrExistingSourceId` has changed to `SourceId`

Still passing in the same CardID and Customer ID to the `SourceId` I get an error "Must provide source or customer."

Here is a code Snippit:
```c#
    var newCharge = new ChargeCreateOptions
                {
                    Amount = (int)(totalAmount * 100),
                    Currency = "usd",
                    CustomerId = _pilotIdentity.Identity.StripeCustomerID,
                    SourceId = CardID,
                    //                    SourceTokenOrExistingSourceId = CardID,
                    Description = "Invoice #: " + invoice.CFIPayInvoiceID,
                    TransferGroup = TransferGroupID,
                    Metadata = new Dictionary<string, string>() {
                        { "Invoice #", invoice.CFIPayInvoiceID.ToString()},
                        {"Student", invoice.InvoiceTo.FirstName + " " + invoice.InvoiceTo.LastName },
                        { "CFI", invoice.Pilot.FirstName + " " + invoice.Pilot.LastName}
                    }
                };
```

**Comment by remi-stripe** [comment]
@flyer299 This error looks different though. This error message usually happens when you don't pass `source` at all or `customer` at all. If you look at the failed request in the dashboard logs you would likely see that `source` is either absent or an empty string indicating that `_pilotIdentity.Identity.StripeCustomerID` and `CardID` are both empty in your code.

You can see that the code is the library is simply
```
        [JsonProperty("source")]
        public string SourceId { get; set; }
```

If that does not help, Can you share more details, especially an example failed request id (req_12345) as found in your dashboard logs?

**Comment by flyer299** [comment]
Sorry for the poor formatting... But here is a dump of my newCharge object that shows the values that go with it.    And Log ID #: req_tSdL0B5qqTw1tE  

As you can see I am sending a valid customer ID and Card ID into the SDK, but the values isn't included in the Request POST Body at all.
  | Name | Value | Type
-- | -- | -- | --
◢ | newCharge | {Stripe.ChargeCreateOptions} | Stripe.ChargeCreateOptions
  | Amount | 5880 | long?
  | ApplicationFee | null | long?
  | Capture | null | bool?
  | Currency | "usd" | string
  | CustomerId | "cus_DlPbleWaCulcVN" | string
  | Description | "Invoice #: 1055" | string
  | Destination | null | Stripe.ChargeDestinationCreateOptions
  | ExchangeRate | null | decimal?
  | ▶ Expand | Count = 0 | System.Collections.Generic.List<string>
  | ▶ ExtraParams | Count = 0 | System.Collections.Generic.Dictionary<string, string>
  | Level3 | null | Stripe.ChargeLevel3Options
  | ▶ Metadata | Count = 3 | System.Collections.Generic.Dictionary<string, string>
  | OnBehalfOf | null | string
  | ReceiptEmail | null | string
  | Shipping | null | Stripe.ChargeShippingOptions
  | SourceCard | null | Stripe.CardCreateNestedOptions
  | SourceId | "card_1DJslDGffmOa79pCZspmnaKb" | string
  | StatementDescriptor | null | string
  | TransferGroup | "Invoice-1055" | string

**Comment by remi-stripe** [comment]
@flyer299 I have tried with the latest version and it works for me so I'm not sure what could be causing this. Can you try the following snippet of code:

```
var chargeOptions = new ChargeCreateOptions()
{
    Amount = 2000,
    Currency = "usd",
    SourceId = "tok_visa"
};

ChargeService chargeService = new ChargeService();

Charge charge = chargeService.Create(chargeOptions);
Console.WriteLine(charge.Id);
```

**Comment by flyer299** [comment]
It looks like you are using a 1 time Token for a Card and not a saved card on a customers account.  

instead of using tok_visa  does it work with something like: card_xyz123

**Comment by remi-stripe** [comment]
@flyer299 it does not matter since both are just strings. I just wanted to give you an exact code that works and confirm that in your current setup it still works. If so then it's a separate bug on the way you create the Charge and we can dig into it. Can you run that code to confirm if it works?

**Comment by flyer299** [comment]
I tried the code snippet you suggested, and I think the Charge worked, then I got to the next part of my code which does a transfer and got this error:  req_jTppHKuuXpkr3j

I am thinking the root cause is that the description I am passing in is too long?   In which case the error I got back on the charge is misleading.     I will try my original code with a shorter description.

**Comment by flyer299** [comment]
I was concerned that the Metadata field might have a null value in it and that was the problem...

But I've checked this is what I am passing into the meta data field:


  | Name | Value | Type
-- | -- | -- | --
◢ | Metadata | Count = 3 | System.Collections.Generic.Dictionary<string, string>
  | ▶ [0] | {[Invoice #, 1054]} | System.Collections.Generic.KeyValuePair<string, string>
  | ▶ [1] | {[Student, ACME Student 1]} | System.Collections.Generic.KeyValuePair<string, string>
  | ▶ [2] | {[CFI, ACME CFI 1]} | System.Collections.Generic.KeyValuePair<string, string>

**Comment by flyer299** [comment]
I think I found the error.   If I Change "Invoice #"  to "Invoice"  it works.   Adding a # sign in the Metadata field appears to cause the problem.

**Comment by remi-stripe** [comment]
Gotcha. Metadata have a limit of 500 characters per value!

**Comment by ob-stripe** [comment]
Hi @flyer299. You were correct, the `#` character was causing the issue. This should be fixed in version 20.4.1.
